Projects/Chat/Sources/Views/ChatScreenViewModel.swift

```diff
@@ -126,16 +129,27 @@ public class ChatScreenViewModel: ObservableObject {
     @MainActor
     private func fetchMessages() async {
         do {
+            let store: ChatStore = globalPresentableStoreContainer.get()
             let chatData = try await chatService.getNewMessages()
             let newMessages = chatData.messages.filterNotAddedIn(list: addedMessagesIds)
             withAnimation {
                 self.messages.append(contentsOf: newMessages)
+
+                if hasNext == nil {
+                    if let failedMessages = store.state.failedMessages[conversationId] {
+                        self.messages.insert(contentsOf: failedMessages.reversed(), at: 0)
+                        addedMessagesIds.append(contentsOf: failedMessages.compactMap({ $0.id }))
+                    }
+                }
+
                 self.messages.sort(by: { $0.sentAt > $1.sentAt })
                 self.lastDeliveredMessage = self.messages.first(where: { $0.sender == .member && $0.remoteId != nil })
             }
             self.banner = chatData.banner
             self.conversationStatus = chatData.conversationStatus ?? .open
+            self.conversationId = chatData.conversationId
```

Comment:
Can we move this to the line just after
let chatData = try await chatService.getNewMessages() ?
We should set it at the moment we fetch it.